lib/web_ui/lib/src/engine/skwasm/skwasm_impl/filters.dart

```diff
@@ -100,33 +108,43 @@ class SkwasmColorFilter {
     SkwasmColorFilter inner,
   ) => SkwasmColorFilter._(colorFilterCompose(outer.handle, inner.handle));
 
-  void dispose() {
-    if (!_isDisposed) {
-      colorFilterDispose(handle);
-      _isDisposed = true;
-    }
-  }
+  static final SkwasmFinalizationRegistry<RawColorFilter> _registry =
+    SkwasmFinalizationRegistry<RawColorFilter>(colorFilterDispose);
```

Comment:
Per-type finalization registry is an interesting choice. Why can't we use one registry for all types? I'm not sure what the trade-offs are actually.

Reply:
Because each type has its own dispose function?

Reply:
Back to the mixin discussion, you could have a `Map<Type, DisposeFunction>`. Then you have one registry, but you need to do a map access on every dispose. 🤷

Reply:
Each handle has a different native disposal function. The alternative would be to have a single finalization registry, but add an extra layer of indirection that wraps the handle and is polymorphic on the dispose method (this is basically what you're doing on the CK side with your Ref objects). For the skwasm case, I found it a little cleaner to just do the per-object finalization registry and not end up with an extra layer of indirection.